### PR TITLE
Directive operands

### PR DESCRIPTION
**Pull request by shakram02**

### CODE REVIEW

File: src/lib/basic_types/formats.rs [hunk 1]
```diff
@@ -7,3 +7,8 @@ pub enum Format {
     Four = 4,
     None,
 }
+
+
+pub fn get_bit_count(format: Format) -> i32 {
```
Comment (oddcoder) [inline src/lib/basic_types/formats.rs:12]: useless function get rid of it

Comment (oddcoder) [reply src/lib/basic_types/formats.rs:12]: and why * 8 ?

File: src/lib/basic_types/instruction_set.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use std::collections::HashMap;
-use basic_types::instruction::{AsmOperand, Instruction};
-use basic_types::formats::Format;
+use basic_types::instruction::AsmOperand;
+use basic_types::formats::{Format, get_bit_count};
```
Comment (oddcoder) [inline src/lib/basic_types/instruction_set.rs:3]: what is this for ?

**Review by oddcoder — CHANGES_REQUESTED** [review]
many irrelevant parts!
this was supposed to add only loc counter and make it build..

File: src/lib/basic_types/formats.rs (hunk 1 above)
Comment (shakram02) [reply src/lib/basic_types/formats.rs:12]: as its name implies, it gets the number of bits in an instruction of a given format, it's not useless and it hides unnecessary details in the code that uses it

**Pull request merged**